src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -47,22 +47,49 @@ public virtual async Task<ExecutionResult> ExecuteAsync(ExecutionContext context
         /// </summary>
         protected abstract Task ExecuteNodeTreeAsync(ExecutionContext context, ObjectExecutionNode rootNode);
 
+        /// <summary>
+        /// Returns the root graph type for the execution -- for a specified schema and operation type.
+        /// </summary>
+        protected virtual IObjectGraphType GetOperationRootType(Document document, ISchema schema, Operation operation)
```

Comment:
Moved from `ExecutionHelper`; no changes.  Could be changed to accept `ExecutionContext` rather than `Document`, `ISchema`, and `Operation` but I did not do so.

Reply:
Let's change to single ExecutionContext parameter.